src/Plugin/formatter/FormatterJson.php

```diff
@@ -98,13 +117,17 @@ protected function extractFieldValues($data) {
       }
       $output[$public_field_name] = $value;
     }
+    if ($this->isCacheEnabled($data)) {
```

Comment:
this duplication is nasty - lets break it down, so formatter implementors don't have to deal with it.